### ISSUE TITLE
Audacity does not load/detect pulseaudio or any audio devices

### ISSUE THREAD

**Issue by xu3s** [termux/termux-packages#10412]
### Problem description

I got this issue where audacity does not detect pulseaudio 
![audacity-err](https://user-images.githubusercontent.com/47383486/166090719-2c837c76-e312-4906-9ac9-55f25d0e44e5.png)

### What steps will reproduce the bug?

Open audacity 

### What is the expected behavior?

Pulseaudio or opensles loaded as audio device

### System information

termux-info:

```
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PID=26202
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://grimler.se/termux-packages-24/ stable main
# x11-repo (sources.list.d/x11.list)
deb https://packages-cf.termux.org/apt/termux-x11 x11 main
# sources.list.d/pointless.list
deb https://its-pointless.github.io/files/24 termux extras
Updatable packages:
All packages up to date
Android version:
9
Kernel build information:
Linux localhost 4.4.153-perf+ #1 SMP PREEMPT Thu Feb 20 17:26:22 CST 2020 aarch64 Android
Device manufacturer:
asus
Device model:
ASUS_X00TD

```

**Comment by xtkoba** [comment]
Honestly this is pretty much expected because PortAudio (the library used by Audacity for audio playback and recording) uses ALSA or JACK, neither of which is available in Termux.

**Comment by rallg** [comment]
Using Audacity, 32-bit Android 11. Audacity can open OGG files, as well as its own native projects. It can export OGG and WAV. I can use command-line ffmpeg to convert mp3 to OGG or WAV, manipulate in Audacity, export to OGG or WAV, then use ffmpeg to convert to mp3. Although I have LAME, Audacity does not recognize the library for import. It appears to use it for export, but I believe that is deceptive, as the exported mp3 fails some tests.

Bottom line: Audacity is useful. Not fully capable, but useful.

**Comment by DLC01** [comment]
> Honestly this is pretty much expected because PortAudio (the library used by Audacity for audio playback and recording) uses ALSA or JACK, neither of which is available in Termux.

there is an version of portaudio which has been patcher to use opensles 
https://github.com/Gundersanne/portaudio_opensles

**Comment by macerg** [comment]
> Honestly this is pretty much expected because PortAudio (the library used by Audacity for audio playback and recording) uses ALSA or JACK, neither of which is available in Termux.

Why not pulseaudio?

**Comment by xtkoba** [comment]
See also #821.

**Comment by truboxl** [comment]
Portaudio now has Pulseaudio support, so probably need to rebuild to verify, see https://github.com/PortAudio/portaudio/pull/336

But I don't see Portaudio releasing new tags anytime soon

**Comment by twaik** [comment]
Fun. Audacity with portaudio w/ SLES support crashes with threading stuff. And with portaudio from latest available commit with pulseaudio support it just hangs. Someone should figure it out.

**Comment by twaik** [comment]
[portaudio.tar.gz](https://github.com/user-attachments/files/17401650/portaudio.tar.gz)
This tarball contains portaudio build script which has both pulse and sles backends. And both are useless. Tested with #21824.
